Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -2226,53 +2224,88 @@ void HandleBreakEventSet (Event[] es, bool dequeuing)
 					Step (StepDepth.Into, StepSize.Min);
 				} else {
 					var args = new TargetEventArgs (etype);
-					args.Process = OnGetProcesses () [0];
+					args.Process = OnGetProcesses ()[0];
 					args.Thread = GetThread (args.Process, current_thread);
 					args.Backtrace = backtrace;
 					args.BreakEvent = breakEvent;
-					
+
 					OnTargetEvent (args);
 				}
 			}
 		}
 
-		public void AddUserAssembly(string userAssembly)
-		{
-			if (userAssemblyNames != null && !userAssemblyNames.Contains(userAssembly))
-				userAssemblyNames.Add(userAssembly);
+		public void AddUserAssembly (string userAssembly)
+		{
+			if (userAssemblyNames != null && !userAssemblyNames.Contains (userAssembly))
+				userAssemblyNames.Add (userAssembly);
 		}
 
 		void HandleAssemblyLoadEvents (AssemblyLoadEvent[] events)
 		{
-			var asm = events [0].Assembly;
+			var asm = events[0].Assembly;
 			if (events.Length > 1 && events.Any (a => a.Assembly != asm))
 				throw new InvalidOperationException ("Simultaneous AssemblyLoadEvent for multiple assemblies");
-
-			OnAssemblyLoaded(asm.Location);
-
-			RegisterAssembly(asm);
+			HandleAssemblyLoaded (asm);
+			RegisterAssembly (asm);
 			bool isExternal;
-			isExternal = !UpdateAssemblyFilters (asm) && userAssemblyNames != null;
-
-			if (outputOptions.ModuleLoaded) { 
+			isExternal = !UpdateAssemblyFilters (asm) && userAssemblyNames != null;
+
+			if (outputOptions.ModuleLoaded) {
 				string flagExt = isExternal ? " [External]" : "";
 				OnDebuggerOutput (false, string.Format ("Loaded assembly: {0}{1}\n", asm.Location, flagExt));
 			}
 		}
 
+		private void HandleAssemblyLoaded(AssemblyMirror asm)
+		{
+			var symbolStatus = string.Empty;
+			var assemblyName = string.Empty;
+			var hasSymbol = false;
+			var name = asm.GetName ();
+			var assemblyObject = asm.GetAssemblyObject ();
+			if (!asm.IsDynamic) {
+				var metaData = asm.GetMetadata ();
+				symbolStatus = metaData.MainModule.HasSymbols == true ? "Symbol loaded" : "Skipped loading symbols";
+				assemblyName = metaData.MainModule.Name;
+				hasSymbol = metaData.MainModule.HasSymbols;
+			} else {
+				symbolStatus = "Skipped loading symbol (dynamic)";
+				assemblyName = "Dynamic assembly";
```

Comment:
General question here, how are we handling translations? I am assuming the above text is being displayed in VS Mac? Are we re-translating these over there?

Just wondering if we should avoid having strings here and instead of an enum for symbol status and then generate the strings on the VS Mac side.

Reply:
We might have to add an additional field to the assembly class to indicate that it's dynamic and maybe something to indicate if symbols were loaded or not. then we can assign these strings in the modules pad.

Reply:
I guess two bool properties here HasSymbols and IsDynamic would be enough. Then we do not need any text to be specified here.